packages/block-library/src/social-links/edit.js

```diff
@@ -59,25 +59,56 @@ export function SocialLinksEdit( props ) {
 	const {
 		iconBackgroundColorValue,
 		iconColorValue,
+		preveIconBackgroundColorValue,
 		openInNewTab,
 		size,
 		layout,
 	} = attributes;
 	const usedLayout = layout || getDefaultBlockLayout( name );
 
 	// Remove icon background color if logos only style selected.
-	const logosOnly =
-		attributes.className?.indexOf( 'is-style-logos-only' ) >= 0;
+	const logosOnly = attributes.className
+		? attributes.className.indexOf( 'is-style-logos-only' ) >= 0
+		: false;
 	useEffect( () => {
 		if ( logosOnly ) {
 			setAttributes( {
 				iconBackgroundColor: undefined,
 				customIconBackgroundColor: undefined,
+				preveIconBackgroundColorValue: iconBackgroundColorValue,
 				iconBackgroundColorValue: undefined,
 			} );
 		}
```

Comment:
```suggestion
		} else {
			// Restore previous background attributes.
			setAttributes( { ...backgroundBackup } );
		}
```
Instead of adding two new effect hooks that restore previous attributes it should work to put in an else clause here to do it.

Reply:
Thank you for the feedback @stokesman. This is my first contribution and I appreciate your guidance. I did the changes in the latest PR.